internal/gitprovider/gitlab.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package gitprovider
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/go-retryablehttp"
+	gitlab "gitlab.com/gitlab-org/api/client-go"
+)
+
+type GitLabProvider struct {
+	Client  *gitlab.Client
+	Project string
+}
+
+func NewGitLabProvider(ctx context.Context, opts Options) (*GitLabProvider, error) {
+	var client *gitlab.Client
+	var glOpts []gitlab.ClientOptionFunc
+
+	host, project, err := parseGitLabURL(opts.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	rtClient := retryablehttp.NewClient()
+	if opts.CertPool != nil {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: opts.CertPool,
+			},
+		}
+		rtClient.HTTPClient.Transport = tr
+	}
+	glOpts = append(glOpts, gitlab.WithHTTPClient(rtClient.HTTPClient))
+
+	if host != "https://gitlab.com" {
```

Comment:
I assume we don't have a way to test this, do we?

Reply:
No we rely on all Flux users running GitHub EE, it's the same logic as in NC/CLI.